lib/spack/spack/util/unparse/unparser.py

```diff
@@ -0,0 +1,1170 @@
+# Copyright (c) 2014-2021, Simon Percivall and Spack Project Developers.
+#
+# SPDX-License-Identifier: Python-2.0
+
+"Usage: unparse.py <path to source file>"
+from __future__ import print_function, unicode_literals
+
+import ast
+import sys
+from contextlib import contextmanager
+
+import six
+from six import StringIO
+
+
+# TODO: if we require Python 3.7, use its `nullcontext()`
+@contextmanager
+def nullcontext():
+    yield
+
+
+# Large float and imaginary literals get turned into infinities in the AST.
```

Comment:
Is there documentation for this behavior in the `ast` or `unparse` modules anywhere?

Reply:
No, but unparser.py is basically an external -- I'm hesitant to change much more about it.

Reply:
Oh, right! Thanks--sounds great to keep as-is!